src/main/java/no/sikt/nva/pubchannels/handler/search/SearchByQueryHandler.java

```diff
@@ -79,13 +79,18 @@ protected PaginatedSearchResult<T> processInput(Void input, RequestInfo requestI
         var searchParameters = SearchParameters.fromRequestInfo(requestInfo);
         var searchResult = searchChannel(searchParameters);
 
+        // Create query params for pagination, with offset and size handled separately
+        var baseQueryParameters = getQueryParams(searchParameters);
+        baseQueryParameters.remove(PAGENO_QUERY_PARAM);
+        baseQueryParameters.remove(PAGECOUNT_QUERY_PARAM);
+
         return PaginatedSearchResult.create(
             constructBaseUri(),
             searchParameters.offset(),
             searchParameters.size(),
             searchResult.pageInformation().totalResults(),
             getHits(constructBaseUri(), searchResult, searchParameters.year()),
-            Map.of(QUERY_PARAM, searchParameters.query(), YEAR_QUERY_PARAM, searchParameters.year())
+            baseQueryParameters
```

Comment:
As mentioned in call, these query parameters are not correct input query params for this endpoint. I see that this is not covered in tests (unfortunately). Suggest writing tests asserting expected `id` and `query parameters` in `PaginatedSearchResult`.

Reply:
This should be resolved now, can you take another look?